python/src/greenbids/tailor/core/app/main.py

```diff
@@ -1,21 +1,46 @@
 import contextlib
+import datetime
 from importlib.metadata import distribution
+import logging
+import os
 
 from fastapi import FastAPI
-
-from greenbids.tailor.core import telemetry
+from greenbids.tailor.core import telemetry, models
 from opentelemetry.instrumentation.fastapi import FastAPIInstrumentor
-from . import resources, profiler
-from .routers import root, healthz
 
+from . import profiler, resources, tasks
+from .routers import healthz, root
+
+_logger = logging.getLogger(__name__)
 
 @contextlib.asynccontextmanager
 async def _lifespan(app: FastAPI):
-    resources.APP_RESOURCES.setup()
+    if resources.get_instance.cache_info().currsize > 0:
+        _logger.warn("A resource object was initialized before app startup")
+    resources.get_instance()
     with profiler.profile():
         yield
 
 
+@tasks.repeat_every(
+    seconds=float(
+        os.environ.get(
+            "GREENBIDS_TAILOR_MODEL_REFRESH_SECONDS",
+        )
+        or datetime.timedelta.max.total_seconds()
```

Comment:
suggestion: You can use dedicated function get env var + fallback to default value `os.getenv("GREENBIDS_TAILOR_MODEL_REFRESH_SECONDS",  datetime.timedelta.max.total_seconds())`